tests/integration/test_enrollment/test_agentd_enrollment.py

```diff
@@ -40,57 +40,77 @@
 tests = load_tests(os.path.join(test_data_path, 'wazuh_enrollment_tests.yaml'))
 configurations_path = os.path.join(test_data_path, 'wazuh_enrollment_conf.yaml')
 configurations = load_wazuh_configurations(configurations_path, __name__)
+host_name = socket.gethostname()
+AGENTD_TIMEOUT = 20
+
+daemons_handler_configuration = {'function': {'daemons': ['wazuh-agentd'], 'ignore_errors': False},
+                                 'module':  {'daemons': ['wazuh-modulesd', 'wazuh-analysisd'], 'ignore_errors': False},
+                                 'configuration': {'daemons': ['wazuh-syscheckd'], 'ignore_errors': False}}
 
-AGENTD_TIMEOUT = 5
 
 # Fixtures
+
 @pytest.fixture(scope='module', params=configurations)
 def get_configuration(request):
     """Get configurations from the module"""
     return request.param
 
 
-@pytest.mark.parametrize('test_case', [case for case in tests])
-def test_agentd_enrollment(set_test_case, configure_socket_listener, configure_environment, create_certificates,
-                           set_keys, set_pass, test_case: list):
+@pytest.fixture(scope='function', params=tests)
+def get_current_test_case(request):
+    """Get current test case from the module"""
+    return request.param
+
+
+@pytest.fixture(scope='function')
+def restart_agentd():
+    try:
+        control_service('restart', daemon='wazuh-agentd')
+    except Exception:
+        pass
```

Comment:
The **try** **except** block is required because some tests check a wrong configurations of the agent.
Thus, in some restart attempts, the daemon will not start, and control_service will rise an error.
Is there a way to avoid this?

Reply:
Implemented a new mechanism into the restart_agent fixture to control when it is expected that the agent fails to start